### ISSUE TITLE
java.lang.NullPointerException when getHttpRequestData() is used n RaygunRequestMessage.cfm - when called from inside a thread

### ISSUE THREAD

**Issue by dominic-cm** [MindscapeHQ/raygun4cfml#61]
I could see the trace message in raygun report.

            var httpRequest = getHttpRequestData();

https://github.com/MindscapeHQ/raygun4cfml/blob/89ef8924229aeea7bc512a85e45c298dd86473a7/src/nz/co/ventego-creative/raygun4cfml/RaygunRequestMessage.cfc#L34

This loggingServie call to raygun send in from within a cf thread.

This is the complete stack trace.
```
"Severity","ThreadID","Date","Time","Application","Message"
"Error","cfthread-7","11/14/24","10:28:44","ukrainedateapplication","Error in RaygunRequestMessage.build() - getHttpRequestData() failed:   {"Message":"","StackTrace":"java.lang.NullPointerException\r\n\tat coldfusion.runtime.CFPage.GetHTTPRequestData(CFPage.java:11572)\r\n\tat coldfusion.runtime.CFPage.GetHTTPRequestData(CFPage.java:11541)\r\n\tat cfRaygunRequestMessage2ecfc55720621$funcBUILD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunRequestMessage.cfc:35)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)\r\n\tat coldfusion.runtime.UDFMethod$ReturnTypeFilter.invoke(UDFMethod.java:553)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:310)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:975)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:696)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:503)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4264)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4227)\r\n\tat cfRaygunMessageDetails2ecfc954199912$funcBUILD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunMessageDetails.cfc:58)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)\r\n\tat coldfusion.runtime.UDFMethod$ReturnTypeFilter.invoke(UDFMethod.java:553)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:310)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:975)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:696)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:503)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4264)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4227)\r\n\tat cfRaygunMessage2ecfc1653083406$funcBUILD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunMessage.cfc:37)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)\r\n\tat coldfusion.runtime.UDFMethod$ReturnTypeFilter.invoke(UDFMethod.java:553)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:310)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:975)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:696)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:503)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4264)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4227)\r\n\tat cfRaygunClient2ecfc1973040920$funcBUILDPAYLOAD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunClient.cfc:168)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)\r\n\tat coldfusion.runtime.UDFMethod$ReturnTypeFilter.invoke(UDFMethod.java:553)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:681)\r\n\tat coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:4919)\r\n\tat coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:4900)\r\n\tat cfRaygunClient2ecfc1973040920$funcSEND.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunClient.cfc:86)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)\r\n\tat coldfusion.runtime.UDFMethod$ReturnTypeFilter.invoke(UDFMethod.java:553)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:681)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:980)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:762)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4175)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4139)\r\n\tat cfLoggingService2ecfc810429403$funcLOGLEVELTORAYGUN.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\com\\cupidmedia\\pulsar\\logging\\LoggingService.cfc:220)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:681)\r\n\tat coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:4919)\r\n\tat coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:4900)\r\n\tat cfLoggingService2ecfc810429403$funcTRACE.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\com\\cupidmedia\\pulsar\\logging\\LoggingService.cfc:146)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:681)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:980)\r\n\tat coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:762)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4175)\r\n\tat coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:4139)\r\n\tat cftest_loggingService2ecfm1633446249$func_CFFUNCCFTHREAD_CFTEST_LOGGINGSERVICE2ECFM16334462491.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\test_loggingService.cfm:16)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:623)\r\n\tat coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:516)\r\n\tat coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:463)\r\n\tat coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:438)\r\n\tat coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:310)\r\n\tat coldfusion.runtime.UDFMethod.invokeCFThread(UDFMethod.java:245)\r\n\tat coldfusion.thread.Task.invokeFunction(Task.java:287)\r\n\tat coldfusion.thread.Task.run(Task.java:140)\r\n\tat coldfusion.scheduling.ThreadPool.run(ThreadPool.java:260)\r\n\tat coldfusion.scheduling.WorkerThread.run(WorkerThread.java:71)\r\n","Suppressed":[],"Type":"java.lang.NullPointerException","TagContext":[{"RAW_TRACE":"\tat cfRaygunRequestMessage2ecfc55720621$funcBUILD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunRequestMessage.cfc:35)","LINE":35,"COLUMN":0,"TEMPLATE":"C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunRequestMessage.cfc","ID":"CF_CFPAGE","TYPE":"CFML"},{"RAW_TRACE":"\tat cfRaygunMessageDetails2ecfc954199912$funcBUILD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunMessageDetails.cfc:58)","LINE":58,"COLUMN":0,"TEMPLATE":"C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunMessageDetails.cfc","ID":"CF_TEMPLATEPROXY","TYPE":"CFML"},{"RAW_TRACE":"\tat cfRaygunMessage2ecfc1653083406$funcBUILD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunMessage.cfc:37)","LINE":37,"COLUMN":0,"TEMPLATE":"C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunMessage.cfc","ID":"CF_TEMPLATEPROXY","TYPE":"CFML"},{"RAW_TRACE":"\tat cfRaygunClient2ecfc1973040920$funcBUILDPAYLOAD.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunClient.cfc:168)","LINE":168,"COLUMN":0,"TEMPLATE":"C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunClient.cfc","ID":"CF_TEMPLATEPROXY","TYPE":"CFML"},{"RAW_TRACE":"\tat cfRaygunClient2ecfc1973040920$funcSEND.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunClient.cfc:86)","LINE":86,"COLUMN":0,"TEMPLATE":"C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\raygun4cfml\\RaygunClient.cfc","ID":"CF_UDFMETHOD","TYPE":"CFML"},{"RAW_TRACE":"\tat cfLoggingService2ecfc810429403$funcLOGLEVELTORAYGUN.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\com\\cupidmedia\\pulsar\\logging\\LoggingService.cfc:220)","LINE":220,"COLUMN":0,"TEMPLATE":"C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\com\\cupidmedia\\pulsar\\logging\\LoggingService.cfc","ID":"CF_TEMPLATEPROXY","TYPE":"CFML"},{"RAW_TRACE":"\tat cfLoggingService2ecfc810429403$funcTRACE.runFunction(C:\\HTML\\users\\556503\\ukrainedatecom\\HTML\\com\\cupi"Severity","ThreadID","Date","Time","Application","Message"

**Comment by TheRealAgentK** [comment]
Thx for reporting @dominic-cm.

I think this problem stems from how specifically Adobe CF treats access to the Http request context from within a thread.

I'll try to replicate that soon and get back to you.

**Comment by TheRealAgentK** [comment]
This is fixed via [71a3cecb0c0ee5651b2967cf2edd1e5a461d6b78](https://github.com/TheRealAgentK/raygun4cfml/commit/71a3cecb0c0ee5651b2967cf2edd1e5a461d6b78) and will be released in `1.7.0` soon

**Comment by gedas20** [comment]
This sounds similar to issue #57
